SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -63,20 +68,21 @@ - (instancetype)init {
 #pragma mark - Background Task
 
 /**
- *  Starts a background task that allows the app to search for accessories and while the app is in the background.
+ *  Starts a background task that allows the app to search for accessories while the app is in the background.
  */
 - (void)sdl_backgroundTaskStart {
     if (self.backgroundTaskId != UIBackgroundTaskInvalid) {
         SDLLogV(@"A background task is already running. No need to start a background task.");
         return;
     }
 
+    __weak typeof(self) weakself = self;
     self.backgroundTaskId = [[UIApplication sharedApplication] beginBackgroundTaskWithName:BackgroundTaskName expirationHandler:^{
         SDLLogD(@"Background task expired");
-        [self sdl_backgroundTaskEnd];
+        [weakself sdl_backgroundTaskEnd];
```

Comment:
This, combined with removing it from `dealloc` would cause the background task to never end if `SDLIAPTransport` is deallocated. If we don't call `sdl_backgroundTaskEnd`, then iOS will discover that we don't end our background tasks and will start preventing us from using background time. We need to call `sdl_backgroundTaskEnd` when we're close to expiring.

We may need to move this to another class that won't be dealloced.

Reply:
Fixed. Moved the background task logic to the lifecycle manager.